gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/ConditionalOnRestGatewayEnabled.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.gateway.rest;
+
+import io.camunda.zeebe.gateway.rest.ConditionalOnRestGatewayEnabled.RestGatewayDisabled;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
+@Documented
+@ConditionalOnWebApplication
+@ConditionalOnMissingBean(value = RestGatewayDisabled.class)
```

Comment:
:thought_balloon: It was easier to model this way:

- The default case is the controllers are enabled always.
- The standalone broker disables them by providing this bean.

However, this only works because beans in the `io.camunda.zeebe.broker` package are loaded before the beans in `io.camunda.zeebe.gateway` :smile: 

But I didn't want this module to know about the specific property and config, since that's none of its concerns.